app/models/nomination.rb

```diff
@@ -24,5 +24,28 @@ def self.rank_by_voter_weight
     return sorted_noms
   end
 
+  # Defines an 'approved nomination' as one having its approved attribute set to 'true' by the Admin
+  def self.approved_by_admin
+    self.where({"approved" => true})
+  end
+
+  # Identifies nominations sharing identical attributes (nominee, role, award_id, theater, show)
+  def self.duplicates
+    self.approved_by_admin.select(:nominee,:role,:award_id,:theater,:show).group(:nominee,:role,:award_id,:theater,:show).select("count(*) AS count").having("count(*) > 1")
+  end
+
+  # Returns a hash with keys of single iterations of duplicate nominations (as found by the above method 'duplicate_noms') and values of that nomination's weight as defined by users' weights (who made the nominations) and how many times that nomination occurs
+  def self.ranked_by_weight
+    ranks = {}
+
+    self.duplicates.each do |nom|
+      # array of all of the weights of all users who submitted nominations for this nominee
+      user_weights = Nomination.where(nominee: nom.nominee, role: nom.role, award_id: nom.award_id, theater: nom.theater, show: nom.show).joins(:user).pluck("users.weight")
+
+      # sum the array. that's the total weight for all of these nominations.
+      ranks[nom] = user_weights.sum * nom.count
+    end
+    ranks.sort_by{|k,v| v}.reverse.to_h
```

Comment:
Unused block argument - `k`. If it's necessary, use `_` or `_k` as an argument name to indicate that it won't be used.